### PR TITLE
added LDAP sync doc

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Added documentation for LDAP group sync. Sync added in the following pulls, this is blocked until all are merged:
- [x] [`rfc2307` schema](https://github.com/openshift/origin/pull/4827)
- [x] [`ad` schema](https://github.com/openshift/origin/pull/4972)
- [x] [`augmented-ad` schema](https://github.com/openshift/origin/pull/5086)
- [x] [`unionNameMapper`](https://github.com/openshift/origin/pull/5100)
- [x] [blacklist](https://github.com/openshift/origin/pull/5108)

/cc @deads2k @liggitt 

@adellape Feedback on formatting, etc, appreciated. This is my first `openshift-docs` PR.

### CODE REVIEW

**Comment by deads2k** [comment]
You need an example with `groupUIDNameMapping` in use.

File: admin_guide/syncing_groups_with_ldap.adoc [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+an administrator to perform a number of different sync types with the same server:
+
+An administrator can sync all groups from the LDAP server into OpenShift:
+----
+$ openshift ex sync-groups --sync-config=config.yaml
+----
+An administrator can sync all groups already in OpenShift that correspond to the server specified
+in the configuration file:
+----
+$ openshift ex sync-groups --type=openshift --sync-config=config.yaml
```
Comment (deads2k) [inline admin_guide/syncing_groups_with_ldap.adoc]: Give a whitelist example like this too.

Comment (stevekuznetsov) [inline admin_guide/syncing_groups_with_ldap.adoc]: I've re-written that section, hopefully for the better. I'm thinking about making a table or something, it's not the clearest command.

File: admin_guide/syncing_groups_with_ldap.adoc [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+this must be the top of your directory tree, but it could also specify a subtree in the directory.
+<2> The scope of the search. Can be either either `base`, `one` or `sub`. If the scope is not
+provided, the default is to use a scope of `sub`. Descriptions of the scope options are given in the
+link:#ldap-search[appendix.]
+<3> The behavior of the search with respect to aliases in the LDAP tree. Can be either `never`,
+`search`, `base` or `always`. If the behavior is not given, the default is to `always` dereference
+aliases. Descriptions of the dereferencing behaviors are given in the
+link:#deref-aliases[appendix].
+<4> The time limit allowed for the search by the client, in seconds. A value of 0 imposes no client-
+side limit.
```
Comment (adellape) [inline admin_guide/syncing_groups_with_ldap.adoc]: Extra space: "client- side"

File: admin_guide/syncing_groups_with_ldap.adoc [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+groups and stored them in an LDAP server. OpenShift allows for you to sync those records with
+internal OpenShift records so that you can focus on managing your groups in once place only.
+OpenShift currently supports group sync with LDAP servers using three common schema for defining
+group membership: RFC2307, Active Directory, and augmented Active Directory.
+
+[IMPORTANT]
+====
+These solutions are a work in progress. As improvements are made,
+these instructions will be simplified.
+====
```
Comment (adellape) [inline admin_guide/syncing_groups_with_ldap.adoc]: Per the above, do you expect things to change any by OSE 3.1?

**Comment by adellape** [comment]
cc @tpoitras since you have the related docs BZ (https://bugzilla.redhat.com/show_bug.cgi?id=1268179).

@stevekuznetsov Do you expect `sync-groups` and `prune-groups` to remain under `openshift ex` for OSE 3.1? Or might they be under `oadm` by then?

**Comment by adellape** [comment]
@stevekuznetsov Sorry, also meant to ask if you still consider this PR [wip].

**Comment by tpoitras** [comment]
@adellape once this merges I'll jump on it as per BZ1268179 :+1: 

@stevekuznetsov Can I be of any assistance?

File: admin_guide/syncing_groups_with_ldap.adoc (hunk 3 above)
Comment (stevekuznetsov) [inline admin_guide/syncing_groups_with_ldap.adoc]: This was stuck in there before dev-cut. @deads2k Do we expect major changes anymore?

**Comment by stevekuznetsov** [comment]
@adellape I do expect the commands to graduate from `openshift ex` soon, and when they do they will live under `oadm`. There was a question of where to put them, but I *think* they will be `oadm groups sync` and `oadm groups prune`. @deads2k Do you know about this?

**Comment by deads2k** [comment]
> @adellape I do expect the commands to graduate from openshift ex soon, and when they do they will live under oadm. There was a question of where to put them, but I think they will be oadm groups sync and oadm groups prune. @deads2k Do you know about this?

You can open a followup pull to move it once we decide.

File: admin_guide/syncing_groups_with_ldap.adoc (hunk 3 above)
Comment (deads2k) [inline admin_guide/syncing_groups_with_ldap.adoc]: > This was stuck in there before dev-cut. @deads2k Do we expect major changes anymore?

Just make sure they are accurate.  We may change how a few things are spelled, but no concepts are changing.

**Comment by stevekuznetsov** [comment]
We will be supporting nested groups better in the future, as well, so, yes, there will be a follow-up. This will not be [wip] once I add the manual mapping today.

**Comment by stevekuznetsov** [comment]
@adellape This is no longer [wip]

**Comment by deads2k** [comment]
Is this still waiting?

**Comment by stevekuznetsov** [comment]
This is waiting on pruning. I can remove that part of the doc and make
another pull to add it of you want.

**Comment by tpoitras** [comment]
I'm not familiar with "pruning". If this work is complete, why are we waiting on merging it here? @stevekuznetsov @deads2k @adellape

**Comment by deads2k** [comment]
> This is waiting on pruning. I can remove that part of the doc and make
> another pull to add it of you want.

remove pruning from this doc, get this merged and open a new on for pruning.

**Comment by adellape** [comment]
> I'm not familiar with "pruning". If this work is complete, why are we waiting on merging it here?

@tpoitras Pruning as in, the `prune-groups` command from the checklist at the top of this PR. That code is still pending in https://github.com/openshift/origin/pull/5145.

@stevekuznetsov LMK when you've removed it and I'll merge!

**Comment by adellape** [comment]
Looks like this has merge conflicts now, presumably from https://github.com/openshift/openshift-docs/pull/1029/files#diff-e612d75a53fc180c5c4512806a7b1c31.

**Comment by stevekuznetsov** [comment]
@adellape I've removed that section.

File: _build_cfg.yml [hunk 4]
```diff
@@ -206,6 +206,8 @@ Topics:
     File: http_proxies
   - Name: Troubleshooting Networking
     File: sdn_troubleshooting
+  - Name: Syncing Groups With syncing_groups_with_ldapP
```
Comment (adellape) [inline _build_cfg.yml]: Oops, just noticed the nav title got messed up here.

**Comment by adellape** [comment]
@stevekuznetsov I'll merge this after the above build_cfg issue is fixed up.

File: _build_cfg.yml (hunk 4 above)
Comment (stevekuznetsov) [inline _build_cfg.yml]: Look like an auto-correct error. I'll update that right away

Comment (stevekuznetsov) [inline _build_cfg.yml]: I do not see this locally ...

**Comment by stevekuznetsov** [comment]
@adellape I can't reproduce that locally, re-pushing hasn't fixed it. It is unclear to me what is up.
screenshot from 2015-11-03 16 45 53

**Comment by stevekuznetsov** [comment]
@adellape  I'm not thinking straight typo fixed.

**Comment by adellape** [comment]
@stevekuznetsov Awesome, thanks!

**Pull request merged**